### PR TITLE
Clean up ORT server routing

### PR DESCRIPTION
**Pull request by csteegz**
**Description**: Describe your changes.

Clean up ORT server routing by joining the regex for the two routes.

**Motivation and Context**
- Why is this change required? What problem does it solve?

This makes explicit the route that ORTServer will use in KFServing

### CODE REVIEW

File: onnxruntime/test/server/unit_tests/http_routes_tests.cc [hunk 1]
```diff
@@ -10,6 +10,7 @@ namespace onnxruntime {
 namespace server {
 namespace test {
 
+static const std::string predict_regex = R"(/v1/models/([^/:]+)(?:/versions/(\d+))?:(classify|regress|predict))";
```
Comment (tmccrmck) [inline onnxruntime/test/server/unit_tests/http_routes_tests.cc]: 👍

Comment (rakelkar) [reply onnxruntime/test/server/unit_tests/http_routes_tests.cc]: looks different from what is registered though?
```
/(?:v1/models/([^/:]+)(?:/versions/(\d+))?:(classify|regress|predict))|score
```

**Comment by rakelkar** [comment]
@vinitra what tag can I use to refer to this image in the container registry?

**Pull request merged**